### PR TITLE
Add emoji picker to chat composer

### PR DESCRIPTION
**Pull request by estrattonbailey**
Many thanks to @nucleartux!

### CODE REVIEW

File: src/view/com/composer/text-input/web/EmojiPicker.web.tsx [hunk 1]
```diff
@@ -26,22 +26,41 @@ export type Emoji = {
...
   isOpen: boolean
-  pos: {top: number; left: number; right: number; bottom: number}
+  pos: EmojiPickerPosition
 }
 
 interface IProps {
   state: EmojiPickerState
   close: () => void
+  /**
+   * If `true`, verrides position and ensures picker is pinned to the top of
```
Comment (surfdude29) [inline src/view/com/composer/text-input/web/EmojiPicker.web.tsx:45]:
```suggestion
   * If `true`, overrides position and ensures picker is pinned to the top of
```
small typo fix

Comment (estrattonbailey) [reply src/view/com/composer/text-input/web/EmojiPicker.web.tsx]: Nice one 🤝

File: src/screens/Messages/Conversation/MessageInput.web.tsx [hunk 2]
```diff
@@ -118,7 +145,23 @@ export function MessageInput({
...
+        <Button
+          onPress={e => {
+            e.currentTarget.measure((_fx, _fy, _width, _height, px, py) => {
+              openEmojiPicker?.({top: py, left: px, right: px, bottom: py})
+            })
+          }}
+          style={[a.p_xs, {marginTop: 3}]}
+          label={_(msg`Open emoji picker`)}
+          accessibilityHint={_(msg`Open emoji picker`)}
+          variant="ghost"
```
Comment (haileyok) [inline src/screens/Messages/Conversation/MessageInput.web.tsx:157]: small nit, can we make the background color of this match the background color of the input? or maybe just some color of gray? it looks kinda loud on bright mode

Screenshot 2024-09-06 at 2 57 22 PM

**Review by haileyok — APPROVED** [review]

File: src/screens/Messages/Conversation/MessageInput.web.tsx (hunk 2 above)
Comment (estrattonbailey) [reply src/screens/Messages/Conversation/MessageInput.web.tsx:157]: Sure I'll try some things

**Comment by estrattonbailey** [comment]
Played around with it and ended up here:
CleanShot 2024-09-06 at 17 42 43@2x
CleanShot 2024-09-06 at 17 42 48@2x
CleanShot 2024-09-06 at 17 42 55@2x
CleanShot 2024-09-06 at 17 42 59@2x

**Comment by haileyok** [comment]
✌️

**Pull request merged**